# Constitution du dataset à partir des données exportées de CEGID expert

Dossiers exportés :
- Missions du cabinet sur 2020
- Facturation 2020
- Entrées et sorties des clients

## Importation des librairies

In [1]:
import pandas as pd
import numpy as np
import re
from scipy import stats

## Lecture de l'ensemble des missions du cabinet sur 2020
1 ligne = 1 mission réalisée par 1 assistant pour le compte d'un client

In [2]:
missions = pd.read_csv("./Data/missions_2020.txt", sep="\t", decimal='.')

C:\Users\tweet\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (8) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
missions.head(10)

Code Mission Mission  Exercice Client   Raison sociale  DEPART CLIENT  \
0  ACOM2020011338400     COM      2020  A0598  OLIVEIRA FAMILY            1.0   
1  ACOM2020011194200     COM      2020  05775        LE MANACH            NaN   
2  ACOM2020011194200     COM      2020  05775        LE MANACH            NaN   
3  AJUR2020011784100     JUR      2020  A0964        L'ATELIER            1.0   
4  AJUR2020011817200     JUR      2020  A0977            JM/MD            NaN   
5  AJUR2020011700700     JUR      2020  A0953            F.PRO            NaN   
6  AJUR2020011700700     JUR      2020  A0953            F.PRO            NaN   
7  AJUR2020011784100     JUR      2020  A0964        L'ATELIER            1.0   
8  ACOM2020011278100     COM      2020  10595            MEIRA            NaN   
9  ACOM2020011278100     COM      2020  10595            MEIRA            NaN   

  Assistant ASSOCIES/EC Article  \
0       MF8         NAB     021   
1       852         SLE     001   
2       852         SLE     001   
3       FK1         STL     069   
4       690         JOP  F06017   
5       VU1         EMG     060   
6       VU1         EMG     060   
7       FK1         STL     069   
8       852         SLE     001   
9       852         SLE     001   

                                             Libellé Qté activité  \
0           DECLARATIONS MENSUELLES / TRIMESTRIELLES         0,25   
1                              RENSEIGNEMENTS CLIENT         0,50   
2                              RENSEIGNEMENTS CLIENT         0,50   
3                                   DIVERS JURIDIQUE         0,25   
4  Secrétariat juridique annuel SCI / SC PORTEFEU...         1,00   
5                       SECRETARIAT JURIDIQUE ANNUEL         0,25   
6                       SECRETARIAT JURIDIQUE ANNUEL         0,25   
7                                   DIVERS JURIDIQUE         0,50   
8                              RENSEIGNEMENTS CLIENT         0,75   
9                              RENSEIGNEMENTS CLIENT         0,25   

  Prix de vente Total  de vente Département        SITE  Unnamed: 15  
0         56,00           14,00       SOCSJ      OLIVET          NaN  
1         63,00           31,50       SOCSJ      OLIVET          NaN  
2         63,00           31,50       SOCSJ      OLIVET          NaN  
3         63,00           15,75         NaN  PITHIVIERS          NaN  
4        170,00          170,00         INT  ANGERVILLE          NaN  
5         74,00           18,50         JUR      YERRES          NaN  
6         74,00           18,50         JUR      YERRES          NaN  
7         63,00           31,50         NaN  PITHIVIERS          NaN  
8         63,00           47,25       SOCSJ      OLIVET          NaN  
9         63,00           15,75       SOCSJ      OLIVET          NaN

In [4]:
#Le dataset des missions 2020 contient 307126 entrées et 16 colonnes
missions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307126 entries, 0 to 307125
Data columns (total 16 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Code Mission     307126 non-null  object 
 1   Mission          307126 non-null  object 
 2   Exercice         307126 non-null  int64  
 3   Client           307126 non-null  object 
 4   Raison sociale   307126 non-null  object 
 5   DEPART CLIENT    18850 non-null   float64
 6   Assistant        307125 non-null  object 
 7   ASSOCIES/EC      307126 non-null  object 
 8   Article          307126 non-null  object 
 9   Libellé          307126 non-null  object 
 10  Qté activité     307126 non-null  object 
 11  Prix de vente    307126 non-null  object 
 12  Total  de vente  307126 non-null  object 
 13  Département      304054 non-null  object 
 14  SITE             307126 non-null  object 
 15  Unnamed: 15      0 non-null       float64
dtypes: float64(2), int64(1), object(13)
me

In [5]:
#Exemple sur un code mission :
missions[missions['Code Mission'] == "ACOM2020012810400"]

Code Mission Mission  Exercice Client             Raison sociale  \
7268    ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
7606    ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
7966    ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
8315    ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
15280   ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
...                   ...     ...       ...    ...                        ...   
175306  ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
176962  ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
176963  ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
177677  ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   
179325  ACOM2020012810400     COM      2020  A1329  ATOUT FLUID CLIMATISATION   

        DEPART CLIENT Assistant ASSOCIES/EC Article  \
7268              NaN       LVN         EMG     012   
7606              NaN       LVN         EMG     012   
7966              NaN       LVN         EMG     012   
8315              NaN       LVN         EMG     012   
15280             NaN       LVN         EMG     012   
...               ...       ...         ...     ...   
175306            NaN       LVN         EMG     013   
176962            NaN       LVN         EMG     013   
176963            NaN       LVN         EMG     013   
177677            NaN       LVN         EMG     013   
179325            NaN       LVN         EMG     011   

                                          Libellé Qté activité Prix de vente  \
7268                          DECLARATIONS T.V.A.         3,00        110,00   
7606                          DECLARATIONS T.V.A.         0,50        110,00   
7966                          DECLARATIONS T.V.A.         3,00        110,00   
8315                          DECLARATIONS T.V.A.         1,50        110,00   
15280                         DECLARATIONS T.V.A.         3,00        110,00   
...                                           ...          ...           ...   
175306  COMPTES ANNUELS  ET DECLARATIONS FISCALES         3,00         94,00   
176962  COMPTES ANNUELS  ET DECLARATIONS FISCALES         2,00         94,00   
176963  COMPTES ANNUELS  ET DECLARATIONS FISCALES         0,50         94,00   
177677  COMPTES ANNUELS  ET DECLARATIONS FISCALES         1,50         94,00   
179325                       CONTROLE ET REVISION         1,00         94,00   

       Total  de vente Département  SITE  Unnamed: 15  
7268            330,00         CPT  EVRY          NaN  
7606             55,00         CPT  EVRY          NaN  
7966            330,00         CPT  EVRY          NaN  
8315            165,00         CPT  EVRY          NaN  
15280           330,00         CPT  EVRY          NaN  
...                ...         ...   ...          ...  
175306          282,00         CPT  EVRY          NaN  
176962          188,00         CPT  EVRY          NaN  
176963           47,00         CPT  EVRY          NaN  
177677          141,00         CPT  EVRY          NaN  
179325           94,00         CPT  EVRY          NaN  

[72 rows x 16 columns]

In [6]:
missions.columns

Index(['Code Mission', 'Mission', 'Exercice', 'Client', 'Raison sociale',
       'DEPART CLIENT', 'Assistant', 'ASSOCIES/EC', 'Article', 'Libellé',
       'Qté activité', 'Prix de vente', 'Total  de vente', 'Département',
       'SITE', 'Unnamed: 15'],
      dtype='object')

In [7]:
#Supprimer la dernière colonne inutile
missions.drop(columns=['Département','Unnamed: 15'], inplace=True)

In [8]:
missions.columns = ['code_mission','mission','exercice','code_client',
                    'client','depart_client','code_assistant','manager','code_article',
                    'libelle_article','temps','cout_horaire','valorisation_temps','site']

In [9]:
missions.dtypes

code_mission           object
mission                object
exercice                int64
code_client            object
client                 object
depart_client         float64
code_assistant         object
manager                object
code_article           object
libelle_article        object
temps                  object
cout_horaire           object
valorisation_temps     object
site                   object
dtype: object

In [10]:
missions.isna().sum()

code_mission               0
mission                    0
exercice                   0
code_client                0
client                     0
depart_client         288276
code_assistant             1
manager                    0
code_article               0
libelle_article            0
temps                      0
cout_horaire               0
valorisation_temps         0
site                       0
dtype: int64

In [11]:
missions[missions['manager'] == 'CHB']

code_mission mission  exercice code_client  \
61      AJUR2020010977600     JUR      2020       01425   
65      AJUR2020010977600     JUR      2020       01425   
84      AJUR2020010977600     JUR      2020       01425   
103     AJUR2020010977600     JUR      2020       01425   
144     AJUR2020010977600     JUR      2020       01425   
7482    AJUR2020010977600     JUR      2020       01425   
57236   AGE02020013247500     GE0      2020       07854   
57852   AGE02020013247500     GE0      2020       07854   
62533   AGE02020013247500     GE0      2020       07854   
63338   AGE02020013247500     GE0      2020       07854   
65670   AGE02020013247500     GE0      2020       07854   
104798  AGE02020013247500     GE0      2020       07854   
120073  AGE02020013247500     GE0      2020       07854   

                             client  depart_client code_assistant manager  \
61               SOCIETE ALVES-CRUZ            4.0            666     CHB   
65               SOCIETE ALVES-CRUZ            4.0            564     CHB   
84               SOCIETE ALVES-CRUZ            4.0            666     CHB   
103              SOCIETE ALVES-CRUZ            4.0            666     CHB   
144              SOCIETE ALVES-CRUZ            4.0            666     CHB   
7482             SOCIETE ALVES-CRUZ            4.0            666     CHB   
57236   AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   
57852   AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   
62533   AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   
63338   AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   
65670   AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   
104798  AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   
120073  AVENIR PAYSAGE ET DIEUDONNE            NaN            065     CHB   

       code_article               libelle_article temps cout_horaire  \
61              069              DIVERS JURIDIQUE  0,25        68,00   
65              069              DIVERS JURIDIQUE  0,25        58,00   
84              069              DIVERS JURIDIQUE  1,00        68,00   
103             060  SECRETARIAT JURIDIQUE ANNUEL  0,50        68,00   
144             060  SECRETARIAT JURIDIQUE ANNUEL  0,25        68,00   
7482            069              DIVERS JURIDIQUE  0,25        65,00   
57236           048                    EVALUATION  1,25       160,00   
57852           048                    EVALUATION  0,50       160,00   
62533           048                    EVALUATION  4,50       165,00   
63338           046  TRANSMISSION DE L'ENTREPRISE  2,50       165,00   
65670           046  TRANSMISSION DE L'ENTREPRISE  0,50       165,00   
104798          046  TRANSMISSION DE L'ENTREPRISE  1,50       165,00   
120073          046  TRANSMISSION DE L'ENTREPRISE  3,00       165,00   

       valorisation_temps    site  
61                  17,00  BRIARE  
65                  14,50  BRIARE  
84                  68,00  BRIARE  
103                 34,00  BRIARE  
144                 17,00  BRIARE  
7482                16,25  BRIARE  
57236              200,00    GIEN  
57852               80,00    GIEN  
62533              742,50    GIEN  
63338              412,50    GIEN  
65670               82,50    GIEN  
104798             247,50    GIEN  
120073             495,00    GIEN

In [12]:
#missions_agg2.columns

In [13]:
#missions_agg2[missions_agg2['manager', '<lambda>'] == 'CHB']

## Modifier les colonnes catégorielles en float

In [14]:
#Remplacer les virgules par des points et espaces pour créer un nombre décimal
missions['temps'] = [x.replace(',', '.') for x in missions['temps']]
missions['temps'] = missions['temps'].astype(float)

In [15]:
missions['cout_horaire'] = [x.replace(',', '.') for x in missions['cout_horaire']]
missions['cout_horaire'] = [re.sub('\s','',x) for x in missions['cout_horaire']]
missions['cout_horaire'] = pd.to_numeric(missions['cout_horaire'])

In [16]:
missions['valorisation_temps'] = [x.replace(',', '.') for x in missions['valorisation_temps']]
missions['valorisation_temps'] = [re.sub('\s','',x) for x in missions['valorisation_temps']]
missions['valorisation_temps'] = pd.to_numeric(missions['valorisation_temps'])

In [17]:
missions

code_mission mission  exercice code_client  \
0       ACOM2020011338400     COM      2020       A0598   
1       ACOM2020011194200     COM      2020       05775   
2       ACOM2020011194200     COM      2020       05775   
3       AJUR2020011784100     JUR      2020       A0964   
4       AJUR2020011817200     JUR      2020       A0977   
...                   ...     ...       ...         ...   
307121  ACOM2020011385400     COM      2020       07264   
307122  ACOM2020011258600     COM      2020       07507   
307123  ACAB2020012991300     CAB      2020       09400   
307124  ACAB2020012991300     CAB      2020       09400   
307125  ACAB2020012991300     CAB      2020       09400   

                                client  depart_client code_assistant  \
0                      OLIVEIRA FAMILY            1.0            MF8   
1                            LE MANACH            NaN            852   
2                            LE MANACH            NaN            852   
3                            L'ATELIER            1.0            FK1   
4                                JM/MD            NaN            690   
...                                ...            ...            ...   
307121                          VEKTIS            NaN           1003   
307122  CABINET DENTAIRE DOCTEUR PAGBE            NaN            134   
307123                          CONGES            NaN            787   
307124                          CONGES            NaN            787   
307125                          CONGES            NaN            787   

               manager code_article  \
0                  NAB          021   
1                  SLE          001   
2                  SLE          001   
3                  STL          069   
4                  JOP       F06017   
...                ...          ...   
307121             NAB           11   
307122             GSO           16   
307123  NON FACTURABLE           98   
307124  NON FACTURABLE           98   
307125  NON FACTURABLE           98   

                                          libelle_article  temps  \
0                DECLARATIONS MENSUELLES / TRIMESTRIELLES   0.25   
1                                   RENSEIGNEMENTS CLIENT   0.50   
2                                   RENSEIGNEMENTS CLIENT   0.50   
3                                        DIVERS JURIDIQUE   0.25   
4       Secrétariat juridique annuel SCI / SC PORTEFEU...   1.00   
...                                                   ...    ...   
307121                               CONTROLE ET REVISION   0.75   
307122                       AUTRES DECLARATIONS FISCALES   0.75   
307123                             CONGES ET RECUPERATION   8.50   
307124                             CONGES ET RECUPERATION   8.50   
307125                             CONGES ET RECUPERATION   8.50   

        cout_horaire  valorisation_temps               site  
0               56.0               14.00             OLIVET  
1               63.0               31.50             OLIVET  
2               63.0               31.50             OLIVET  
3               63.0               15.75         PITHIVIERS  
4              170.0              170.00         ANGERVILLE  
...              ...                 ...                ...  
307121          66.0               49.50  LA FERTE ST AUBIN  
307122          66.0               49.50       VILLEMANDEUR  
307123          89.5              760.75     NON FACTURABLE  
307124          89.5              760.75     NON FACTURABLE  
307125          89.5              760.75     NON FACTURABLE  

[307126 rows x 14 columns]

## Agrégation du dataframe sur le code_mission pour sommer le temps et la valorisation du temps

Il faut garder une version non aggrégée pour avoir :
- les différents codes assistants par manager
- Les différents sites
- Les différents code_article et libellé articles (activités réalisées)

In [18]:
#Exemple sur un code mission :
missions[missions['code_mission'] == "ACOM2020012810400"].code_article.unique()

array(['012', '010', '016', '011', '001', '04950', '005', '019', '013'],
      dtype=object)

In [19]:
#missions_agg3.iloc[('AAGS2020013017300', '001'),:]

In [20]:
missions.columns

Index(['code_mission', 'mission', 'exercice', 'code_client', 'client',
       'depart_client', 'code_assistant', 'manager', 'code_article',
       'libelle_article', 'temps', 'cout_horaire', 'valorisation_temps',
       'site'],
      dtype='object')

In [21]:
missions.code_mission.values

array(['ACOM2020011338400', 'ACOM2020011194200', 'ACOM2020011194200', ...,
       'ACAB2020012991300', 'ACAB2020012991300', 'ACAB2020012991300'],
      dtype=object)

In [22]:
missions_agg = missions.groupby(['code_mission']).agg({'mission':[lambda x:x.value_counts().index[0]],
                                                        'exercice':[lambda x:x.value_counts().index[0]],
                                                        'code_client': [lambda x:x.value_counts().index[0]],
                                                        'code_assistant': ['nunique'],
                                                        'manager': [lambda x:x.value_counts().index[0]],
                                                        'code_article': [lambda x: list(x.unique())],
                                                        'libelle_article': [lambda x: list(x.unique())],
                                                        'temps': ['sum'],
                                                        'valorisation_temps': ['sum'],
                                                        'site': [lambda x:x.value_counts().index[0]]})

In [23]:
missions_agg.head()

mission exercice code_client code_assistant  manager  \
                  <lambda> <lambda>    <lambda>        nunique <lambda>   
code_mission                                                              
AAGS2020013017300      AGS     2020       06565              4      FRL   
AAGS2020013017600      AGS     2020       06924              3      JOP   
ACAB2020012807600      CAB     2020       08937              5      CIT   
ACAB2020012807700      CAB     2020       08949              2      CIT   
ACAB2020012807800      CAB     2020       08952              8      CIT   

                                                        code_article  \
                                                            <lambda>   
code_mission                                                           
AAGS2020013017300                                         [001, 021]   
AAGS2020013017600                                         [001, 029]   
ACAB2020012807600                     [092, 089, 078, 083, 081, 079]   
ACAB2020012807700                        [081, 078, 079, 0794, 0793]   
ACAB2020012807800  [077, 076, 085, 078, 079, 081, 088, 0796, 0794...   

                                                     libelle_article   temps  \
                                                            <lambda>     sum   
code_mission                                                                   
AAGS2020013017300  [RENSEIGNEMENTS CLIENT, DECLARATIONS MENSUELLE...   15.00   
AAGS2020013017600  [RENSEIGNEMENTS CLIENT, AUTRES TRAVAUX EN SOCIAL]   20.75   
ACAB2020012807600  [FORMATION PROFESSIONNELLE INTERNE, ASSISTANCE...   75.75   
ACAB2020012807700  [COMPTABILITE GENERALE CABINET, FACTURATION ET...  113.75   
ACAB2020012807800  [SECRETARIAT CABINET, GESTION DES TEMPS, NON F...  207.25   

                  valorisation_temps               site  
                                 sum           <lambda>  
code_mission                                             
AAGS2020013017300            1194.51               GIEN  
AAGS2020013017600            1747.40         PITHIVIERS  
ACAB2020012807600            5793.25  DOSSIERS INTERNES  
ACAB2020012807700           17637.76  DOSSIERS INTERNES  
ACAB2020012807800           26408.25  DOSSIERS INTERNES

In [24]:
missions_agg.loc['ACOM2020012810400',:]

mission             <lambda>                                                  COM
exercice            <lambda>                                                 2020
code_client         <lambda>                                                A1329
code_assistant      nunique                                                     6
manager             <lambda>                                                  EMG
code_article        <lambda>      [012, 010, 016, 011, 001, 04950, 005, 019, 013]
libelle_article     <lambda>    [DECLARATIONS T.V.A., SAISIE ET CENTRALISATION...
temps               sum                                                    141.75
valorisation_temps  sum                                                   13758.5
site                <lambda>                                                 EVRY
Name: ACOM2020012810400, dtype: object

In [25]:
missions_agg.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9928 entries, AAGS2020013017300 to ATB 2020014179000
Data columns (total 10 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   (mission, <lambda>)          9928 non-null   object 
 1   (exercice, <lambda>)         9928 non-null   int64  
 2   (code_client, <lambda>)      9928 non-null   object 
 3   (code_assistant, nunique)    9928 non-null   int64  
 4   (manager, <lambda>)          9928 non-null   object 
 5   (code_article, <lambda>)     9928 non-null   object 
 6   (libelle_article, <lambda>)  9928 non-null   object 
 7   (temps, sum)                 9928 non-null   float64
 8   (valorisation_temps, sum)    9928 non-null   float64
 9   (site, <lambda>)             9928 non-null   object 
dtypes: float64(2), int64(2), object(6)
memory usage: 1.1+ MB


## Lecture du dossier facturation sur 2020
Quel coût facturé au client ?

In [26]:
facturation = pd.read_csv("./Data/facturation_2020.txt", sep="\t")

In [27]:
facturation.head()

Code mission Client  Exercice Mission  Total HT  Exercice.1  \
0  ACOM2020012810400  A1329    2020.0     COM  1 530,00      2020.0   
1  ACOM2020012810700  A1330    2020.0     COM  1 524,00      2020.0   
2  ACOM2020012810800  A1331    2020.0     COM    578,00      2020.0   
3  ACOM2020012811700  A1337    2020.0     COM    484,00      2020.0   
4  ACOM2020012811800  A1338    2020.0     COM  1 294,00      2020.0   

         Date Nature  N° Pièce  Article  Qté facturée Prix unitaire  \
0  20/10/2019    FAC       4.0  ACOMPTE           1.0      1 530,00   
1  20/10/2019    FAC       5.0  ACOMPTE           1.0      1 524,00   
2  20/10/2019    FAC       6.0  ACOMPTE           1.0        578,00   
3  20/10/2019    FAC      12.0  ACOMPTE           1.0        484,00   
4  20/10/2019    FAC      13.0  ACOMPTE           1.0      1 294,00   

   Numéro d'ordre                Désignation  \
0             1.0  Acompte Provisionnel N° 7   
1             1.0  Acompte Provisionnel N° 7   
2             1.0  Acompte Provisionnel N° 5   
3             1.0  Acompte Provisionnel N° 4   
4             1.0  Acompte Provisionnel N° 4   

                     Raison sociale  Unnamed: 15  
0         ATOUT FLUID CLIMATISATION          NaN  
1                        ATOUTRENOV          NaN  
2                            AUX 3M          NaN  
3                            DECAME          NaN  
4  DESIGN FONCTIONNALITE INGENIERIE          NaN

In [28]:
facturation.dtypes

Code mission       object
Client             object
Exercice          float64
Mission            object
Total HT           object
Exercice.1        float64
Date               object
Nature             object
N° Pièce          float64
Article            object
Qté facturée      float64
Prix unitaire      object
Numéro d'ordre    float64
Désignation        object
Raison sociale     object
Unnamed: 15       float64
dtype: object

In [29]:
#Supprimer cette ligne qui est nulle et empêche le pd.to_numeric
facturation.drop(732, axis=0, inplace=True)

In [30]:
facturation.isnull().sum()

Code mission          0
Client                0
Exercice              0
Mission               0
Total HT              0
Exercice.1            0
Date                  0
Nature                0
N° Pièce              0
Article               0
Qté facturée          0
Prix unitaire         0
Numéro d'ordre        0
Désignation           6
Raison sociale        1
Unnamed: 15       52103
dtype: int64

In [31]:
# Remplacer les valeurs catégorielles par des floats
facturation['Total HT'] = [str(x).replace(',', '.') for x in facturation['Total HT']]
facturation['Total HT'] = [re.sub('\s','',x) for x in facturation['Total HT']]
facturation['Total HT'] = pd.to_numeric(facturation['Total HT'])

In [32]:
facturation['Total HT'].isnull().sum()

0

### Remplir les valeurs NaN par 0 dans le Total HT de la facturation 

In [33]:
facturation.fillna(0, inplace=True)

In [34]:
facturation['Qté facturée'].value_counts().head(10)

1.0     40809
2.0      3062
3.0      1988
4.0      1232
5.0       815
6.0       619
0.0       499
7.0       437
8.0       284
12.0      275
Name: Qté facturée, dtype: int64

In [35]:
#Vérification que le client 00030 a bien les mêmes données que sur le logiciel
facturation.query('Client == "00030"')

Code mission Client  Exercice Mission  Total HT  Exercice.1  \
51935  ACAC2020012636500  00030    2020.0     CAC    515.00      2020.0   
51936  ACAC2020012636500  00030    2020.0     CAC      2.58      2020.0   
51987  ACAC2020012636500  00030    2020.0     CAC   1442.00      2020.0   
51988  ACAC2020012636500  00030    2020.0     CAC      7.21      2020.0   
52035  ACAC2020012636500  00030    2020.0     CAC   4326.00      2020.0   
52036  ACAC2020012636500  00030    2020.0     CAC     21.63      2020.0   
52100  ACAC2020012636500  00030    2020.0     CAC   1442.00      2020.0   
52101  ACAC2020012636500  00030    2020.0     CAC      7.21      2020.0   

             Date Nature  N° Pièce  Article  Qté facturée Prix unitaire  \
51935  23/06/2020    FAC     816.0      221           1.0        515,00   
51936  23/06/2020    FAC     816.0      H3C           1.0        515,00   
51987  25/02/2021    FAC     943.0  ACOMPTE           1.0      1 442,00   
51988  25/02/2021    FAC     943.0      H3C           1.0      1 442,00   
52035  29/04/2021    FAC     982.0  ACOMPTE           1.0      4 326,00   
52036  29/04/2021    FAC     982.0      H3C           1.0      4 326,00   
52100  23/07/2021    FAC    1018.0  ACOMPTE           1.0      1 442,00   
52101  23/07/2021    FAC    1018.0      H3C           1.0      1 442,00   

       Numéro d'ordre                    Désignation  \
51935             1.0  Attestations ventes de grumes   
51936             2.0                 Cotisation H3C   
51987             1.0                   Acompte N° 1   
51988             2.0                 Cotisation H3C   
52035             1.0                   Acompte N° 2   
52036             2.0                 Cotisation H3C   
52100             1.0                   Acompte N° 3   
52101             2.0                 Cotisation H3C   

                    Raison sociale  Unnamed: 15  
51935  ETABLISSEMENTS HENRY MILLET          0.0  
51936  ETABLISSEMENTS HENRY MILLET          0.0  
51987  ETABLISSEMENTS HENRY MILLET          0.0  
51988  ETABLISSEMENTS HENRY MILLET          0.0  
52035  ETABLISSEMENTS HENRY MILLET          0.0  
52036  ETABLISSEMENTS HENRY MILLET          0.0  
52100  ETABLISSEMENTS HENRY MILLET          0.0  
52101  ETABLISSEMENTS HENRY MILLET          0.0

## Agrégation du dataset facturation sur le code_mission pour sommer le total HT facturé au client sur une mission particulière

In [36]:
fact_agg = facturation.groupby(['Code mission']).agg({'Total HT':['sum']})

In [37]:
fact_agg.head()

Total HT
                       sum
Code mission              
AAGS2020013017300   1350.0
AAGS2020013017600   2350.0
AANT2020012595000   7689.5
AANT2020013216300   8256.0
AANT2020013546600    510.0

Il existe plus de missions dans facturation que de missions dans le dataset mission ?

## Concaténation du dataset facturation sur l'ensemble des missions

In [38]:
data = pd.concat([missions_agg, fact_agg], axis=1)

In [39]:
data.columns = ['mission','exercice','code_client','code_assistant','manager','code_article',
                'libelle_article','temps','valorisation_temps','site','facturation']

In [40]:
#508 factures à 0 euro
data.isna().sum()

mission               691
exercice              691
code_client           691
code_assistant        691
manager               691
code_article          691
libelle_article       691
temps                 691
valorisation_temps    691
site                  691
facturation           508
dtype: int64

In [41]:
len(data.index)

10619

In [42]:
#Remplacer NaN par 0 euro
data['facturation'].fillna(0, inplace=True)

In [43]:
#Le reste des valeurs nulles correspondent au facture sur des codes_missions qui n'existent pas dans le dataset missions
data.isna().sum()

mission               691
exercice              691
code_client           691
code_assistant        691
manager               691
code_article          691
libelle_article       691
temps                 691
valorisation_temps    691
site                  691
facturation             0
dtype: int64

Il se peut que des missions ne soient pas terminées sur le millésime 2020 mais qu'il existe une facture

Supprimer les 691 lignes ajoutées sans aucune donnée car une facture mais pas dans de code_mission dans le dataset missions

In [44]:
list_index = data[data['code_client'].isna()].index
len(list_index)

691

In [45]:
data.drop(list_index, axis=0, inplace=True)

In [46]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9928 entries, AAGS2020013017300 to ATB 2020014179000
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   mission             9928 non-null   object 
 1   exercice            9928 non-null   float64
 2   code_client         9928 non-null   object 
 3   code_assistant      9928 non-null   float64
 4   manager             9928 non-null   object 
 5   code_article        9928 non-null   object 
 6   libelle_article     9928 non-null   object 
 7   temps               9928 non-null   float64
 8   valorisation_temps  9928 non-null   float64
 9   site                9928 non-null   object 
 10  facturation         9928 non-null   float64
dtypes: float64(5), object(6)
memory usage: 930.8+ KB


In [47]:
data.head()

mission  exercice code_client  code_assistant manager  \
AAGS2020013017300     AGS    2020.0       06565             4.0     FRL   
AAGS2020013017600     AGS    2020.0       06924             3.0     JOP   
ACAB2020012807600     CAB    2020.0       08937             5.0     CIT   
ACAB2020012807700     CAB    2020.0       08949             2.0     CIT   
ACAB2020012807800     CAB    2020.0       08952             8.0     CIT   

                                                        code_article  \
AAGS2020013017300                                         [001, 021]   
AAGS2020013017600                                         [001, 029]   
ACAB2020012807600                     [092, 089, 078, 083, 081, 079]   
ACAB2020012807700                        [081, 078, 079, 0794, 0793]   
ACAB2020012807800  [077, 076, 085, 078, 079, 081, 088, 0796, 0794...   

                                                     libelle_article   temps  \
AAGS2020013017300  [RENSEIGNEMENTS CLIENT, DECLARATIONS MENSUELLE...   15.00   
AAGS2020013017600  [RENSEIGNEMENTS CLIENT, AUTRES TRAVAUX EN SOCIAL]   20.75   
ACAB2020012807600  [FORMATION PROFESSIONNELLE INTERNE, ASSISTANCE...   75.75   
ACAB2020012807700  [COMPTABILITE GENERALE CABINET, FACTURATION ET...  113.75   
ACAB2020012807800  [SECRETARIAT CABINET, GESTION DES TEMPS, NON F...  207.25   

                   valorisation_temps               site  facturation  
AAGS2020013017300             1194.51               GIEN       1350.0  
AAGS2020013017600             1747.40         PITHIVIERS       2350.0  
ACAB2020012807600             5793.25  DOSSIERS INTERNES          0.0  
ACAB2020012807700            17637.76  DOSSIERS INTERNES          0.0  
ACAB2020012807800            26408.25  DOSSIERS INTERNES       1700.0

## Ajouter les entrées et sorties des clients MCCM

In [48]:
clients = pd.read_csv("./Data/client_dates.csv", sep=";")

In [49]:
clients_formes = pd.read_csv("./Data/client-naf-forme.csv", sep=";")

In [50]:
clients.head()

Code Date création Départ cabinet
0   30    07/11/1995              0
1   44    31/10/1995              0
2  441    30/10/1995              0
3  455    07/11/1995              0
4  549    09/11/1995              0

In [51]:
clients.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11281 entries, 0 to 11280
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Code            11281 non-null  object
 1   Date création   11277 non-null  object
 2   Départ cabinet  11281 non-null  object
dtypes: object(3)
memory usage: 264.5+ KB


In [52]:
clients.dtypes

Code              object
Date création     object
Départ cabinet    object
dtype: object

In [53]:
def entree_sortie(col):
    
    for date in str(col):
        if '2020' in str(col):
            return 1
        else:
            return 0

In [54]:
#Test de la fonction
entree_sortie("20/02/2020")

1

In [55]:
clients['entrée_clt'] = clients['Date création'].apply(entree_sortie)

In [56]:
clients['sortie_clt'] = clients['Départ cabinet'].apply(entree_sortie)

In [57]:
clients.head(50)

Code Date création Départ cabinet  entrée_clt  sortie_clt
0     30    07/11/1995              0           0           0
1     44    31/10/1995              0           0           0
2    441    30/10/1995              0           0           0
3    455    07/11/1995              0           0           0
4    549    09/11/1995              0           0           0
5    569    06/11/1995              0           0           0
6    579    06/11/1995     31/03/2020           0           1
7    584    10/11/1995              0           0           0
8    613    16/11/1995              0           0           0
9    632    30/10/1995              0           0           0
10   656    06/11/1995              0           0           0
11   666    03/11/1995              0           0           0
12   676    31/10/1995              0           0           0
13   685    03/11/1995              0           0           0
14   689    15/02/1996              0           0           0
15   707    03/11/1995              0           0           0
16   713    30/10/1995     15/12/2020           0           1
17   795    31/10/1995              0           0           0
18   952    06/11/1995              0           0           0
19   961    06/11/1995              0           0           0
20  1065    31/10/1995              0           0           0
21  1078    03/11/1995              0           0           0
22  1088    03/11/1995              0           0           0
23  1169    30/10/1995              0           0           0
24  1221    31/10/1995              0           0           0
25  1259    31/10/1995              0           0           0
26  1318    30/10/1995              0           0           0
27  1349    06/11/1995              0           0           0
28  1350    31/10/1995              0           0           0
29  1353    31/10/1995              0           0           0
30  1369    10/11/1995              0           0           0
31  1395    31/10/1995              0           0           0
32  1398    06/11/1995              0           0           0
33  1412    31/10/1995              0           0           0
34  1413    10/11/1995              0           0           0
35  1420    08/01/1996              0           0           0
36  1426    31/10/1995              0           0           0
37  1440    08/11/1995              0           0           0
38  1467    06/11/1995              0           0           0
39  1476    28/02/1996              0           0           0
40  1489    03/11/1995              0           0           0
41  1498    06/11/1995              0           0           0
42  1509    17/11/1995              0           0           0
43  1534    31/10/1995              0           0           0
44  1570    31/10/1995              0           0           0
45  1590    08/11/1995              0           0           0
46  1602    03/11/1995              0           0           0
47  1626    31/10/1995              0           0           0
48  1629    06/11/1995              0           0           0
49  1650    05/04/1996              0           0           0

In [58]:
clients.columns

Index(['Code', 'Date création', 'Départ cabinet', 'entrée_clt', 'sortie_clt'], dtype='object')

In [59]:
clients.columns = ['code_client', 'ENTREE_CLIENT', 'SORTIE_CLIENT', 'entrée_clt', 'sortie_clt']

In [60]:
clients.code_client.value_counts()

3408      1
3528      1
A2344     1
FORM5     1
7061      1
         ..
2856      1
A2690     1
A3378     1
BERTON    1
7979      1
Name: code_client, Length: 11281, dtype: int64

In [61]:
clients_formes.head()

Code                                           Code NAF Forme juridique
0    9                                             Divers              SA
1   25                          Autre imprimerie (labeur)            SARL
2   26                   Édition de revues et périodiques              SA
3   30                  Fabrication de parquets assemblés             SAS
4   36  Commerce de détail d'habillement en magasin sp...            SARL

## Concaténer les entrées et les sorties au dataset selon le code_client avec la fonction merge()

In [62]:
clients.code_client

0           30
1           44
2          441
3          455
4          549
         ...  
11276    P7129
11277    P7503
11278    P7540
11279    P7673
11280    P7911
Name: code_client, Length: 11281, dtype: object

In [63]:
len(clients.code_client[0])

2

In [64]:
def suppr_0(s):
    return s.lstrip("0")

In [65]:
#Test de la suppression de 0 au début des identifiants
suppr_0("06565")

'6565'

In [66]:
data.code_client = data.code_client.apply(suppr_0)

In [67]:
data.head()

mission  exercice code_client  code_assistant manager  \
AAGS2020013017300     AGS    2020.0        6565             4.0     FRL   
AAGS2020013017600     AGS    2020.0        6924             3.0     JOP   
ACAB2020012807600     CAB    2020.0        8937             5.0     CIT   
ACAB2020012807700     CAB    2020.0        8949             2.0     CIT   
ACAB2020012807800     CAB    2020.0        8952             8.0     CIT   

                                                        code_article  \
AAGS2020013017300                                         [001, 021]   
AAGS2020013017600                                         [001, 029]   
ACAB2020012807600                     [092, 089, 078, 083, 081, 079]   
ACAB2020012807700                        [081, 078, 079, 0794, 0793]   
ACAB2020012807800  [077, 076, 085, 078, 079, 081, 088, 0796, 0794...   

                                                     libelle_article   temps  \
AAGS2020013017300  [RENSEIGNEMENTS CLIENT, DECLARATIONS MENSUELLE...   15.00   
AAGS2020013017600  [RENSEIGNEMENTS CLIENT, AUTRES TRAVAUX EN SOCIAL]   20.75   
ACAB2020012807600  [FORMATION PROFESSIONNELLE INTERNE, ASSISTANCE...   75.75   
ACAB2020012807700  [COMPTABILITE GENERALE CABINET, FACTURATION ET...  113.75   
ACAB2020012807800  [SECRETARIAT CABINET, GESTION DES TEMPS, NON F...  207.25   

                   valorisation_temps               site  facturation  
AAGS2020013017300             1194.51               GIEN       1350.0  
AAGS2020013017600             1747.40         PITHIVIERS       2350.0  
ACAB2020012807600             5793.25  DOSSIERS INTERNES          0.0  
ACAB2020012807700            17637.76  DOSSIERS INTERNES          0.0  
ACAB2020012807800            26408.25  DOSSIERS INTERNES       1700.0

In [68]:
data[data['code_client'] == '6087SAS'].index

Index(['ACOM2020011207000', 'ASOP2020013263300'], dtype='object')

In [69]:
data['code_client'].replace(['6087SAS'], '06087SAS', inplace=True)

In [70]:
data.reset_index(inplace=True)

In [71]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9928 entries, 0 to 9927
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   index               9928 non-null   object 
 1   mission             9928 non-null   object 
 2   exercice            9928 non-null   float64
 3   code_client         9928 non-null   object 
 4   code_assistant      9928 non-null   float64
 5   manager             9928 non-null   object 
 6   code_article        9928 non-null   object 
 7   libelle_article     9928 non-null   object 
 8   temps               9928 non-null   float64
 9   valorisation_temps  9928 non-null   float64
 10  site                9928 non-null   object 
 11  facturation         9928 non-null   float64
dtypes: float64(5), object(7)
memory usage: 930.9+ KB


In [72]:
data_2020 = pd.merge(data, clients[['code_client','entrée_clt', 'sortie_clt']], how='left', on="code_client")

In [73]:
data_2020.head()

index mission  exercice code_client  code_assistant manager  \
0  AAGS2020013017300     AGS    2020.0        6565             4.0     FRL   
1  AAGS2020013017600     AGS    2020.0        6924             3.0     JOP   
2  ACAB2020012807600     CAB    2020.0        8937             5.0     CIT   
3  ACAB2020012807700     CAB    2020.0        8949             2.0     CIT   
4  ACAB2020012807800     CAB    2020.0        8952             8.0     CIT   

                                        code_article  \
0                                         [001, 021]   
1                                         [001, 029]   
2                     [092, 089, 078, 083, 081, 079]   
3                        [081, 078, 079, 0794, 0793]   
4  [077, 076, 085, 078, 079, 081, 088, 0796, 0794...   

                                     libelle_article   temps  \
0  [RENSEIGNEMENTS CLIENT, DECLARATIONS MENSUELLE...   15.00   
1  [RENSEIGNEMENTS CLIENT, AUTRES TRAVAUX EN SOCIAL]   20.75   
2  [FORMATION PROFESSIONNELLE INTERNE, ASSISTANCE...   75.75   
3  [COMPTABILITE GENERALE CABINET, FACTURATION ET...  113.75   
4  [SECRETARIAT CABINET, GESTION DES TEMPS, NON F...  207.25   

   valorisation_temps               site  facturation  entrée_clt  sortie_clt  
0             1194.51               GIEN       1350.0           0           1  
1             1747.40         PITHIVIERS       2350.0           0           0  
2             5793.25  DOSSIERS INTERNES          0.0           0           0  
3            17637.76  DOSSIERS INTERNES          0.0           0           0  
4            26408.25  DOSSIERS INTERNES       1700.0           0           0

In [74]:
clients_formes.columns

Index(['Code', 'Code NAF', 'Forme juridique'], dtype='object')

In [75]:
clients_formes.columns = ['code_client', 'Code NAF', 'Forme juridique']

In [76]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9928 entries, 0 to 9927
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   index               9928 non-null   object 
 1   mission             9928 non-null   object 
 2   exercice            9928 non-null   float64
 3   code_client         9928 non-null   object 
 4   code_assistant      9928 non-null   float64
 5   manager             9928 non-null   object 
 6   code_article        9928 non-null   object 
 7   libelle_article     9928 non-null   object 
 8   temps               9928 non-null   float64
 9   valorisation_temps  9928 non-null   float64
 10  site                9928 non-null   object 
 11  facturation         9928 non-null   float64
 12  entrée_clt          9928 non-null   int64  
 13  sortie_clt          9928 non-null   int64  
dtypes: float64(5), int64(2), object(7)
memory usage: 1.1+ MB


In [77]:
data_final_2020 = pd.merge(data_2020, clients_formes[['code_client', 'Code NAF','Forme juridique']], how='left', on="code_client")

In [78]:
data_final_2020.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9929 entries, 0 to 9928
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   index               9929 non-null   object 
 1   mission             9929 non-null   object 
 2   exercice            9929 non-null   float64
 3   code_client         9929 non-null   object 
 4   code_assistant      9929 non-null   float64
 5   manager             9929 non-null   object 
 6   code_article        9929 non-null   object 
 7   libelle_article     9929 non-null   object 
 8   temps               9929 non-null   float64
 9   valorisation_temps  9929 non-null   float64
 10  site                9929 non-null   object 
 11  facturation         9929 non-null   float64
 12  entrée_clt          9929 non-null   int64  
 13  sortie_clt          9929 non-null   int64  
 14  Code NAF            9929 non-null   object 
 15  Forme juridique     9929 non-null   object 
dtypes: flo

In [79]:
#Vérification des valeurs nulles
data_final_2020.isna().sum()

index                 0
mission               0
exercice              0
code_client           0
code_assistant        0
manager               0
code_article          0
libelle_article       0
temps                 0
valorisation_temps    0
site                  0
facturation           0
entrée_clt            0
sortie_clt            0
Code NAF              0
Forme juridique       0
dtype: int64

In [80]:
data_final_2020.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9929 entries, 0 to 9928
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   index               9929 non-null   object 
 1   mission             9929 non-null   object 
 2   exercice            9929 non-null   float64
 3   code_client         9929 non-null   object 
 4   code_assistant      9929 non-null   float64
 5   manager             9929 non-null   object 
 6   code_article        9929 non-null   object 
 7   libelle_article     9929 non-null   object 
 8   temps               9929 non-null   float64
 9   valorisation_temps  9929 non-null   float64
 10  site                9929 non-null   object 
 11  facturation         9929 non-null   float64
 12  entrée_clt          9929 non-null   int64  
 13  sortie_clt          9929 non-null   int64  
 14  Code NAF            9929 non-null   object 
 15  Forme juridique     9929 non-null   object 
dtypes: flo

In [81]:
#Vérification de combien de lignes par client
data_final_2020.code_client.value_counts()

7436     10
4794      9
77254     9
6459      8
40056     8
         ..
3696      1
A1123     1
441       1
A3254     1
A2669     1
Name: code_client, Length: 5091, dtype: int64

In [82]:
data_final_2020[data_final_2020['code_client']=='7436']

index mission  exercice code_client  code_assistant manager  \
622   ACOM2020011087000     COM    2020.0        7436             5.0     EMG   
3832  AGE02020012886600     GE0    2020.0        7436             4.0     EMG   
4157  AGEJ2020013003600     GEJ    2020.0        7436             3.0     EMG   
4628  AGEP2020013003800     GEP    2020.0        7436             6.0     EMG   
4721  AGJ02020013414300     GJ0    2020.0        7436             2.0     EMG   
4869  AGS02020013215600     GS0    2020.0        7436             5.0     EMG   
5364  AGS12020014007700     GS1    2020.0        7436             2.0     EMG   
5474  AGS32020013349100     GS3    2020.0        7436             4.0     EMG   
6035  AJUR2020011008800     JUR    2020.0        7436             5.0     EMG   
9759  ASOP2020013168500     SOP    2020.0        7436            12.0     EMG   

                                           code_article  \
622   [016, 010, 001, 018, 04950, 011, 005, 019, 013...   
3832                                         [001, 018]   
4157                                          [069, 69]   
4628                                              [045]   
4721                                              [069]   
4869                                         [025, 001]   
5364                                              [024]   
5474                                          [029, 29]   
6035                                         [069, 060]   
9759  [020, F0200, F02083, 021, 029, F0206, 001, 049...   

                                        libelle_article   temps  \
622   [AUTRES DECLARATIONS FISCALES, SAISIE ET CENTR...  115.75   
3832   [RENSEIGNEMENTS CLIENT, SITUATION INTERMEDIAIRE]   34.00   
4157                                 [DIVERS JURIDIQUE]    8.50   
4628                    [ASSISTANCE AU CONTROLE URSSAF]   28.75   
4721                                 [DIVERS JURIDIQUE]    5.25   
4869       [CONTRATS DE TRAVAIL, RENSEIGNEMENTS CLIENT]    9.25   
5364                       [FORMALITES DE LICENCIEMENT]    7.50   
5474                         [AUTRES TRAVAUX EN SOCIAL]   52.50   
6035   [DIVERS JURIDIQUE, SECRETARIAT JURIDIQUE ANNUEL]    9.75   
9759  [FICHE DE PAIE, BULLETIN DE PAIE, FORFAIT MENS...  406.75   

      valorisation_temps           site  facturation  entrée_clt  sortie_clt  \
622             11372.42  FONTAINEBLEAU     16900.00           0           0   
3832             2915.63  FONTAINEBLEAU      2000.00           0           0   
4157              602.25  FONTAINEBLEAU       310.00           0           0   
4628             2328.13  FONTAINEBLEAU      2793.41           0           0   
4721              443.25  FONTAINEBLEAU       620.00           0           0   
4869              942.00  FONTAINEBLEAU       350.00           0           0   
5364              561.00  FONTAINEBLEAU      1560.00           0           0   
5474             8238.63  FONTAINEBLEAU      5610.00           0           0   
6035              821.00  FONTAINEBLEAU      1220.00           0           0   
9759            13047.29  FONTAINEBLEAU      6472.00           0           0   

                                               Code NAF Forme juridique  
622   Commerce de gros (commerce interentreprises) d...            SASU  
3832  Commerce de gros (commerce interentreprises) d...            SASU  
4157  Commerce de gros (commerce interentreprises) d...            SASU  
4628  Commerce de gros (commerce interentreprises) d...            SASU  
4721  Commerce de gros (commerce interentreprises) d...            SASU  
4869  Commerce de gros (commerce interentreprises) d...            SASU  
5364  Commerce de gros (commerce interentreprises) d...            SASU  
5474  Commerce de gros (commerce interentreprises) d...            SASU  
6035  Commerce de gros (commerce interentreprises) d...            SASU  
9759  Commerce de gros (commerce interentreprises) d...            SASU

In [84]:
#Supprimer les dossiers internes qui sont pour le cabinet et non facturable
data_final_2020[data_final_2020['mission'] == "SST"].index

Int64Index([9914, 9915], dtype='int64')

In [85]:
data_final_2020.drop([9914, 9915], axis=0, inplace=True)

In [92]:
#drop une ligne qui est un doublon
data_final_2020.drop(8160, axis=0, inplace=True)

In [93]:
data_final_2020.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9926 entries, 0 to 9928
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   code_mission        9926 non-null   object 
 1   type_mission        9926 non-null   object 
 2   exercice            9926 non-null   float64
 3   code_client         9926 non-null   object 
 4   total_assistant     9926 non-null   float64
 5   manager             9926 non-null   object 
 6   code_article        9926 non-null   object 
 7   libelle_article     9926 non-null   object 
 8   temps               9926 non-null   float64
 9   valorisation_temps  9926 non-null   float64
 10  site                9926 non-null   object 
 11  facturation         9926 non-null   float64
 12  entrée_clt          9926 non-null   int64  
 13  sortie_clt          9926 non-null   int64  
 14  secteur             9926 non-null   object 
 15  forme               9926 non-null   object 
dtypes: flo

## Exportation du dataset 2020

In [87]:
data_final_2020.columns = ['code_mission', 'type_mission', 'exercice', 'code_client', 'total_assistant',
                           'manager', 'code_article', 'libelle_article', 'temps',
                           'valorisation_temps', 'site', 'facturation', 'entrée_clt', 'sortie_clt', 'secteur', 'forme']

In [94]:
data_final_2020[data_final_2020.code_mission == "ASAP2020013405100"].iloc[:,:15]

code_mission type_mission  exercice code_client  total_assistant  \
8161  ASAP2020013405100          SAP    2020.0       A2541              1.0   

     manager code_article                 libelle_article  temps  \
8161     GSO        [016]  [AUTRES DECLARATIONS FISCALES]   1.75   

      valorisation_temps  site  facturation  entrée_clt  sortie_clt secteur  
8161              124.25  GIEN        155.0           1           0  Divers

In [95]:
data_final_2020.head()

code_mission type_mission  exercice code_client  total_assistant  \
0  AAGS2020013017300          AGS    2020.0        6565              4.0   
1  AAGS2020013017600          AGS    2020.0        6924              3.0   
2  ACAB2020012807600          CAB    2020.0        8937              5.0   
3  ACAB2020012807700          CAB    2020.0        8949              2.0   
4  ACAB2020012807800          CAB    2020.0        8952              8.0   

  manager                                       code_article  \
0     FRL                                         [001, 021]   
1     JOP                                         [001, 029]   
2     CIT                     [092, 089, 078, 083, 081, 079]   
3     CIT                        [081, 078, 079, 0794, 0793]   
4     CIT  [077, 076, 085, 078, 079, 081, 088, 0796, 0794...   

                                     libelle_article   temps  \
0  [RENSEIGNEMENTS CLIENT, DECLARATIONS MENSUELLE...   15.00   
1  [RENSEIGNEMENTS CLIENT, AUTRES TRAVAUX EN SOCIAL]   20.75   
2  [FORMATION PROFESSIONNELLE INTERNE, ASSISTANCE...   75.75   
3  [COMPTABILITE GENERALE CABINET, FACTURATION ET...  113.75   
4  [SECRETARIAT CABINET, GESTION DES TEMPS, NON F...  207.25   

   valorisation_temps               site  facturation  entrée_clt  sortie_clt  \
0             1194.51               GIEN       1350.0           0           1   
1             1747.40         PITHIVIERS       2350.0           0           0   
2             5793.25  DOSSIERS INTERNES          0.0           0           0   
3            17637.76  DOSSIERS INTERNES          0.0           0           0   
4            26408.25  DOSSIERS INTERNES       1700.0           0           0   

                                          secteur forme  
0                        Activités d'architecture  EURL  
1  Fabrication de meubles de bureau et de magasin   SAS  
2  Conseil en systèmes et logiciels informatiques  EURL  
3                            Activités comptables  SASU  
4                            Activités comptables   SAS

In [96]:
data_final_2020.to_csv("./Data/dataset-2020.csv", index=False)